src/run/handlers/tags-handler.cts

```diff
@@ -25,6 +25,27 @@ async function getTagRevalidatedAt(
   return tagManifest.revalidatedAt
 }
 
+/**
+ * Get the most recent revalidation timestamp for a list of tags
+ */
+export async function getMostRecentTagRevalidationTimestamp(tags: string[]) {
+  if (tags.length === 0) {
+    return 0
+  }
+
+  const cacheStore = getMemoizedKeyValueStoreBackedByRegionalBlobStore({ consistency: 'strong' })
+
+  const timestampsOrNulls = await Promise.all(
+    tags.map((tag) => getTagRevalidatedAt(tag, cacheStore)),
+  )
+
+  const timestamps = timestampsOrNulls.filter((timestamp) => timestamp !== null)
+  if (timestamps.length === 0) {
+    return 0
+  }
+  return Math.max(...timestamps)
+}
+
```

Comment:
To support interface of `use cache` handler we need method that will return most recent tag expiration timestamp .. or `0` if none of tags were ever revalidated (as opposed to just checking if tag is stale that is used in response cache handler)